### ISSUE TITLE
Interpolation of get_env cannot contain other functions

### ISSUE THREAD

**Issue by dmlemos** [gruntwork-io/terragrunt#306]
```
# terraform.tfvars

terragrunt = {
  terraform {
    extra_arguments "global_vars" {
      commands = [
        "plan",
        "import",
        "push",
        "refresh",
        "destroy",
      ]

      required_var_files = [
        "${get_env("VARS_FILE", "")}"
      ]
    }
  }
}
```

```sh
export VARS_FILE="\${get_tfvars_dir()}/../../common.tfvars"
terragrunt plan
```

```sh
...
[terragrunt] 2017/10/04 16:39:12 Invalid interpolation syntax. Expected syntax of the form '${function_name()}', but got '${get_tfvars_dir()}'
[terragrunt] 2017/10/04 16:39:12 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

**Comment by brikis98** [comment]
Out of curiosity, does it work if the value contains double quotes?

```
export VARS_FILE='"${get_tfvars_dir()}/../../common.tfvars"'
```

**Comment by dmlemos** [comment]
That produces a different error actually:

```sh
[terragrunt] 2017/10/05 18:26:28 At 19:11: expected '/' for comment
[terragrunt] 2017/10/05 18:26:28 Unable to determine underlying exit code, so Terragrunt will exit with error code 1
```

**Comment by brikis98** [comment]
lol, that's not a very helpful error message. Not sure what the issue is, but clearly Terragrunt's somewhat hacky way of handling interpolations is failing here. 

What is your use case, BTW? Why pass in something via env var that then needs to be dynamically interpolated?

**Comment by dmlemos** [comment]
The reason was to try making #305 work

**Comment by brikis98** [comment]
This should have been fixed as part of the upgrade in [v0.19.0](https://github.com/gruntwork-io/terragrunt/releases/tag/v0.19.0) to support Terraform 0.12 and HCL2.
